pkg/mockclient/mockclient.go

```diff
@@ -122,6 +122,9 @@ func (ms *MockEC2Client) RunInstances(input *ec2.RunInstancesInput) (*ec2.Reserv
 	newInstance := ec2.Instance{
 		InstanceId:     &instanceID,
 		PrivateDnsName: &privateDNSName,
+		NetworkInterfaces: []*ec2.InstanceNetworkInterface{{
+			NetworkInterfaceId: aws.String("i-123456789"),
+		}},
```

Comment:
why was this change needed?

Reply:
This change is needed as during unit tests we would crash here: https://github.com/gardener/machine-controller-manager-provider-aws/blob/ed01605d61036b3bd5e81917c1ee7998f5db5127/pkg/aws/core.go#L251
However, if we inline `networkInterfaceID` as suggested by @ScheererJ, we can remove it again.